### PR TITLE
test: Add CLI tests - pass

### PR DESCRIPTION
**Pull request by dheles**
Thank you for contributing to KBSecret! Please fill out the items below to help us merge
your work more quickly.

- [X] Have you run `make test` locally and ensured that all tests pass?
- [X] Have you run `make coverage` locally and ensured that coverage did not drop?

*If* you're changing something in the library:
- [ ] Have you introduced unit tests for your changes?

*If* you're changing something in the CLI:
- [ ] Have you updated the manual pages and shell completion (if necessary)?
- [X] Have you introduced CLI unit tests for your changes and ensured that `make test-cli` passes?

Please describe your changes briefly here. Thank you!

Add test for `kbsecret pass` cli command

### CODE REVIEW

File: test/cli/test_pass.rb [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    flags = cmd.output
+  end
+
+  # ensure no errors returned when the command is run with each of its flags
+  flags.lines.each do |flag|
+    run_command "kbsecret pass #{label} #{flag}" do |cmd|
+      cmd.wait
+      puts cmd
+      assert_empty cmd.stderr
+    end
```
Comment (woodruffw) [inline test/cli/test_pass.rb]: Make sure to remove the `test-login` record in an `ensure` block here.

File: test/cli/test_pass.rb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      cmd.wait
+      assert_match error_pattern, cmd.output
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+end
+
+def test_flags
```
Comment (woodruffw) [inline test/cli/test_pass.rb]: I think we can drop this test -- `-c`, `--clipboard` won't work properly on the CI anyways, and the rest of the flags aren't unique or featured enough to be worth testing.

File: test/cli/test_pass.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+  def test_session
+    session = "test-session"
+    default_label = "test-default-login"
+    session_label = "test-session-login"
+    default_username = "default_user"
+    default_password = "default_pass"
+    session_username = "session_user"
+    session_password = "session_pass"
+    default_output = "#{default_password}"
```
Comment (woodruffw) [inline test/cli/test_pass.rb]: Unnecessary interpolation:

```ruby
default_output = default_password
session_output = session_password
```

File: test/cli/test_pass.rb [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    run_command "kbsecret pass #{label}" do |cmd|
+      cmd.wait
+      assert_equal output, cmd.output.chomp
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_session
```
Comment (woodruffw) [inline test/cli/test_pass.rb]: `test_pass_with_session` would be a bit more descriptive.

File: test/cli/test_pass.rb [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    end
+
+    # fail to retreive default login from session:
+    run_command "kbsecret pass -s #{session} #{default_label}" do |cmd|
+      cmd.wait
+      assert_match error_pattern_no_record, cmd.stderr
+    end
+
+    # remove session:
+    run_command_and_stop "kbsecret session rm -d #{session}"
```
Comment (woodruffw) [inline test/cli/test_pass.rb]: Put this in the `ensure` block, and run the lines below against a completely made up session instead.

Comment (dheles) [reply test/cli/test_pass.rb]: see #52

File: test/cli/test_pass.rb (hunk 4 above)
Comment (woodruffw) [reply test/cli/test_pass.rb]: Got it. I didn't realize that you were naming the methods after options -- that makes a lot more sense. That's fine then!

**Pull request closed without merging**